src/openassetio-core-c/include/openassetio/c/InfoDictionary.h

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2013-2022 The Foundry Visionmongers Ltd
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+#include <openassetio/c/export.h>
+
+#include "./StringView.h"
+#include "./errors.h"
+#include "./namespace.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+/**
+ * @addtogroup CAPI C API
+ * @{
+ */
+
+/**
+ * Opaque handle type representing a @fqref{InfoDictionary} "InfoDictionary"
+ * instance.
+ *
+ * The associated @fqcref{InfoDictionary_s} "InfoDictionary suite" functions
+ * operate on this handle type.
+ */
+// NOLINTNEXTLINE(modernize-use-using)
+typedef struct OPENASSETIO_NS(InfoDictionary_t) * OPENASSETIO_NS(InfoDictionary_h);
+
+/**
+ * Function pointer suite for a @fqref{InfoDictionary} "InfoDictionary" C API.
+ *
+ * Instances of this suite are provided by the `InfoDictionary_suite`
+ * function and operate on a @fqcref{InfoDictionary_h} "InfoDictionary handle".
+ */
+// NOLINTNEXTLINE(modernize-use-using)
+typedef struct {
+  /**
+   * Destructor function.
+   *
+   * This should be called when the handle is no longer in use.
```

Comment:
[question] Do we need to describe the fact that this will immediately delete the object and so understanding ownership is critical?